testing.rb

```diff
@@ -0,0 +1,389 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/BlockNesting,Metrics/MethodLength,Metrics/ModuleLength,Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity,Metrics/AbcSize
+
+# module Enumerable
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+    arr = self if self.class == Hash
+    n = 0
+    while n < arr.length
+      yield(arr[n])
+      n += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+    arr = self if self.class == Hash
+    n = 0
+    while n < arr.length
+      yield(arr[n], n)
+      n += 1
+    end
+    self
+  end
+
+  def my_none?(input = nil)
+    return true if !block_given? && empty?
+
+    if !block_given? && input == Float
+      my_each do |item|
+        return false if item.class == input
+      end
+    elsif !block_given? && input == String
+      my_each do |item|
+        return false if item.class == input
+      end
+      true
+    elsif !block_given? && input == Numeric
+      my_each do |item|
+        return false if item.class == input
+      end
+      true
+    elsif !block_given? && input.class == Regexp
+      my_each  do |item|
+        char = item.split('')
+        char.my_each do |n|
+          return false if n.match?(input)
+        end
+      end
+      true
+    elsif !block_given? && input.nil?
+      my_each do |item|
+        return false if item == true
+      end
+      true
+    elsif block_given?
+      my_each do |item|
+        return false if item == true
+        return false if yield(item)
+      end
+      true
+    end
+  end
```

Comment:
- when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
- #my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern